File: cpp/121_best_time_to_buy_and_sell_stock.cpp
#include <vector>

using namespace std;

class Solution {
 public:
  int maxProfit(vector<int>& prices) {
    int profit = 0;
    int buy = prices[0];
    for (const auto price : prices) {
      if (price < buy) {
        buy = price;
      } else {
        profit = max(profit, price - buy);
      }
    }

    return profit;
  }
};

File: rust/121_best_time_to_buy_and_sell_stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut profit = 0;
        let mut buy = prices[0];

        for price in prices {
            if price < buy {
                buy = price;
            } else {
                profit = std::cmp::max(profit, price - buy);
            }
        }

        profit
    }
}
